googletest/include/gtest/gtest-printers.h

```diff
@@ -364,11 +364,18 @@ class UniversalPrinter;
 template <typename T>
 void UniversalPrint(const T& value, ::std::ostream* os);
 
+enum DefaultPrinterType {
```

Comment:
This technique is fine, but I'd prefer a few tag types here instead of an enum and WrapPrinterType.
```
struct ContainerPrintTag{};
struct PointerPrintTag{};
struct FunctionPointerPrintTag{};
struct OtherPrintTag{};
```